MovieApp/PresentationLayer/Presenters/HomeScreenPresenter.swift

```diff
@@ -1,11 +1,35 @@
 class HomeScreenPresenter {
 
-    private let movies = MockDataService.movies
+    private var movies: [MovieViewModel] = []
+    private let movieClient: MovieClient
+
+    var delegate: HomeScreenViewController!
+
+    init(baseUrl: String) {
+        movieClient = MovieClient(baseApiClient: BaseApiClient(baseUrl: baseUrl))
+    }
 
     public var numberOfMovies: Int {
         movies.count
     }
 
+    func fetchMovies() {
```

Comment:
How will the VC know that the presenter has finished loading? You need to pass a completion handler to this function, something like:
`func fetchMovies(_ completion: @escaping Result<[MovieViewModel], Error) -> Void)`

and call it when it finishes loading.

Reply:
Did you mean for me to keep the fetched movies in the viewController or in the Presenter?
Because if I'm using a completion function like the one that you wrote it seems convenient to simply save the data in the view controller.

Reply:
Save it in VC.